src/lib.rs

```diff
@@ -198,7 +198,12 @@ pub mod built_info {
 
 /// Returns the host name of the current system.
 pub fn get_hostname() -> std::io::Result<String> {
-    Ok(hostname::get()?.to_string_lossy().into())
+    let hostname = option_env!("VECTOR_HOSTNAME").unwrap_or("");
+    Ok(if !hostname.is_empty() {
+        hostname.to_string()
+    } else {
+        hostname::get()?.to_string_lossy().into()
+    })
```

Comment:
I'd suggest using `VECTOR_HOSTNAME` if set, even if empty, and only fallback to `hostname::get()` if `VECTOR_HOSTNAME` is unset.